### PR TITLE
clears the e2e-test "TestE2E/PAR/ACTIONS/shell/ShellHostname" FAIL

### PR DESCRIPTION
**Pull request by TomokiUemoto**
## Description of the Pull Request (PR):

This PR clears the e2e-test "TestE2E/PAR/ACTIONS/shell/ShellHostname" FAIL.

                --- FAIL: TestE2E/PAR/ACTIONS/shell/ShellHostname (46.63s)
                    actions.go:321: Running command "/usr/local/bin/singularity shell /tmp/stest.841046438/test.sif"
                    singularitycmd.go:363:
                        Console output: hostname^M

                        Expected output: sstk-tx2c1
                    singularitycmd.go:363: error while reading from the console: read |0: i/o timeout
                        checking console output
                        github.com/sylabs/singularity/e2e/internal/e2e.ConsoleExpect.func1
                                github.com/sylabs/singularity@/e2e/internal/e2e/singularitycmd.go:228
                        github.com/sylabs/singularity/e2e/internal/e2e.ConsoleRun.func1
                                github.com/sylabs/singularity@/e2e/internal/e2e/singularitycmd.go:363
                        github.com/sylabs/singularity/e2e/internal/e2e.TestEnv.RunSingularity.func1
                                github.com/sylabs/singularity@/e2e/internal/e2e/singularitycmd.go:622
                        testing.tRunner
                                testing/testing.go:909
                        runtime.goexit
                                runtime/asm_arm64.s:1128

### This fixes or addresses the following GitHub issues:

 - Fixes #

#### Before submitting a PR, make sure you have done the following:

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: e2e/actions/actions.go [hunk 1]
```diff
@@ -302,7 +302,7 @@ func (c actionTests) actionShell(t *testing.T) {
 			argv: []string{c.env.ImagePath},
 			consoleOps: []e2e.SingularityConsoleOp{
 				e2e.ConsoleSendLine("hostname"),
-				e2e.ConsoleExpect(hostname),
+				e2e.ConsoleExpect("hostname"),
```
Comment (dtrudg) [inline e2e/actions/actions.go:305]: This seems to be expecting the `hostname` command to be echoed back, but it is no longer checking that the command returns the hostname that we expect.

**Review by dtrudg — CHANGES_REQUESTED** [review]

**Comment by dtrudg** [comment]
@TomokiUemoto - this change seems to defeat the purpose of the test, which is to check that the hostname is being reported properly from inside the container.

Instead of making this change, it would be good to understand where you see the test fail. What operating system / configuration etc. are you using when the test fails? It isn't an issue we see in our automated testing, or manual testing across various Linux distributions.

Thanks!

**Comment by TomokiUemoto** [comment]
The environment where the test was executed is as follows.
We are connecting to an external network via a proxy server.

[root@sstk-tx2c1 ~]# dmidecode |grep Product
        Product Name: R181-T92-00
        Product Name: MT91-FS4-00
[root@sstk-tx2c1 ~]# cat /etc/redhat-release
Red Hat Enterprise Linux release 8.2 (Ootpa)
[root@sstk-tx2c1 ~]# uname -r
4.18.0-193.el8.aarch64
[root@sstk-tx2c1 ~]# singularity version
3.6.3
[root@sstk-tx2c1 ~]#

Please tell me the possible factors.
  - Does not support aarch64.
  - Not assuming a Proxy environment.
  - Other

**Comment by dtrudg** [comment]
> Please tell me the possible factors.
> 
>     * Does not support aarch64.
> 
>     * Not assuming a Proxy environment.
> 
>     * Other

I have run the test suite successfully on aarch64 machines. It is possible that the issue is a result of the e2e test not supporting a proxy environment, leading to an incorrect test image.

If you are running with a proxy then the e2e tests will need to be modified so that the proxy environment is set, used, and passed consistently. This is not currently on our roadmap but would be welcome as a PR. The content of tests (such as the hostname check here) should not need to be modified if the proxy is used correctly throughout.

**Comment by TomokiUemoto** [comment]
Thanks to your brilliant comment and great tips.
I understand that this issue is due to Proxy setting.
Proxy settings are also common in terms of security.
Therefore, we will create a new PR and request support for the proxy environment.

**Pull request closed without merging**